src/lib/components/data-graphic/compositions/timestamp-detail/TimestampSpark.svelte

```diff
@@ -0,0 +1,175 @@
+<script lang="ts">
+  /**
+   * TimestampSpark.svelte
+   * ---------------------
+   * This simple component is a basic sparkline, meant to be used
+   * in a table / model profile preview.
+   * It optionally enables the user to determine a "window", which
+   * is just a box emcompassing the zoomWindowXMin and zoomWindowXMax values.
+   */
+  import { guidGenerator } from "$lib/util/guid";
+  import { fade } from "svelte/transition";
+  import { cubicOut as easing } from "svelte/easing";
+  import { scaleLinear } from "d3-scale";
+  import { extent } from "d3-array";
+  import { writable } from "svelte/store";
+  import { createExtremumResolutionStore } from "../../extremum-resolution-store";
+  import { lineFactory, areaFactory } from "./utils";
+  import { tweened } from "svelte/motion";
+
+  const plotID = guidGenerator();
+
+  export let data;
+
+  export let width = 360;
+  export let height = 120;
+  export let curve = "curveLinear";
+  export let area = false;
+  export let color = "hsl(217, 10%, 50%)";
+  export let tweenIn = false;
+
+  // the color of the zoom window
+  export let zoomWindowColor = "hsla(217, 90%, 60%, .2)";
+  // the color of the zoom window boundaries
+  export let zoomWindowBoundaryColor = "rgb(100,100,100)";
+  export let zoomWindowXMin: any = undefined;
+  export let zoomWindowXMax: any = undefined;
+
+  export let xAccessor: string = undefined;
+  export let yAccessor: string = undefined;
+
+  // rowsize for table
+  export let left = 4;
+  export let right = 4;
+  export let top = 12;
+  export let bottom = 4;
+
+  export let buffer = 4;
+
+  const X = writable(undefined);
+  const Y = writable(undefined);
+
+  $: plotTop = top + buffer;
+  $: plotBottom = height - buffer - bottom;
+  $: plotLeft = left + buffer;
+  $: plotRight = width - right - buffer;
+
+  // establish basis values
+  let xExtents = extent(data, (d) => d[xAccessor]);
+  $: xExtents = extent(data, (d) => d[xAccessor]);
```

Comment:
Yeah, in this case, it would, but unfortunately we need the values immediately in the component initialization in the `createExtremumResolutionStore` call.